### PR TITLE
Signl4 destination

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

Create doc for Signl4 destination

### CODE REVIEW

**Review by ujjwal-ab — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+Once you have confirmed that the source platform supports sending events to SIGNL4, follow these steps:
+
+1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **SIGNL4**.
+2. Assign a name to the destination and click **Continue**.
+
+## Connection settings
+
+To successfully configure SIGNL4 as a destination, you will need to configure the following settings:
+
+- **API Key**: Enter the team secret of your SIGNL4 team. Refer to the <a href="#faq">FAQ</a> section to know about how to obtain the team secret.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:35]:
```suggestion
- **API Key**: Enter the team secret of your SIGNL4 team. Refer to the <a href="#faq">FAQ</a> section for more information on obtaining the team secret.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- **X-S4-Service**: Enter a category for the alert, for example, IT. Refer to the <Link to="#signl-categories">Signl categories</Link> section to know more about these categories.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:36]:
```suggestion
- **X-S4-Service**: Enter a category for the alert, for example, `IT`. Refer to the <Link to="#signl-categories">Signl categories</Link> section to know more about these categories.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+<img src="../../assets/event-stream-destinations/signl4-connection-settings-2.png" alt="SIGNL4 connection setting"/>
+- **X-S4-Location**: Enter the location in `latitude, longitude` format to display the correct location on the map in the mobile app.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:38]:
```suggestion
- **X-S4-Location**: Enter the location in the `latitude, longitude` format to display the correct location on the map in your mobile app.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+<img src="../../assets/event-stream-destinations/signl4-connection-settings.png" alt="SIGNL4 connection setting"/>
+- **X-S4-Status**: Enter the alert status by choosing the **new**, **acknowledged**, or **resolved** values.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:40]:
```suggestion
- **X-S4-Status**: Enter the alert status by choosing from the **new**, **acknowledged**, or **resolved** values.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+<img src="../../assets/event-stream-destinations/signl4-conn-settings-3.png" alt="SIGNL4 connection setting"/>
+- **X-S4-AlertingScenario**: Enter a value to control how SIGNL4 notifies the team:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:42]:
```suggestion
- **X-S4-AlertingScenario**: Use this setting to control how SIGNL4 notifies the team. RudderStack gives you the following options:
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  - single_ack: Only one person needs to acknowledge the alert.
+  - multi_ack: All on-duty persons need to acknowledge the alert. 
+  - emergency: All persons, irrespective of their duty status, are notified and need to acknowledge the alert.
+<img src="../../assets/event-stream-destinations/signl4-conn-settings-2.png" alt="SIGNL4 connection setting"/>
+- **X-S4-ExternalID**: Use this field to pass the unique ID of a record if an event originates from that record in a third-party system. 
+<img src="../../assets/event-stream-destinations/signl4-conn-settings-5.png" alt="SIGNL4 connection setting"/>
+
+As can be observed, RudderStack dashboard provides the following fields for each of the above-mentioned settings:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:50]:
```suggestion
The RudderStack dashboard contains the following fields for each of the above-mentioned settings:
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+- **Default Value**: Enter the actual value for the field.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:52]:
```suggestion
- **Default Value**: Enter or select the required value for the field.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+- **RudderStack property name**: Enter a property name whose actual value will be taken from the payload and assigned to the field.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:53]:
```suggestion
- **RudderStack property name**: Enter the property name whose actual value will be taken from the event payload and assigned to the field.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+Although these fields are optional, a higher priority is given to the **RudderStack property name** if both are specified. For example:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:55]:
```suggestion
Although these fields are optional, RudderStack prioritizes **RudderStack property name** over **Default Value** if both are specified. For example, consider the following use-cases:
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+- If you provide `new` in the **Default value** field for **X-S4-Status**, the output payload will contain `X-S4-Status: new`.
+- If you provide `alert_status` in the **RudderStack property name** field, then the **X-S4-Status** value is taken from `properties.alert_status`. Suppose the value for `properties.alert_status` is set to `resolved`, the output payload will contain `X-S4-Status: resolved`.
+
+If you provide values in both fields, the final output payload will contain `X-S4-Status: resolved`, as the **RudderStack property name** is given the higher priority.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:59]:
```suggestion
- If you provide values in both the fields, the final output payload sent to SIGNL4 will contain `X-S4-Status: resolved`, as the **RudderStack property name** is given higher priority over **Default Field**.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+<img src="../../assets/event-stream-destinations/signl4-conn-settings-5.png" alt="SIGNL4 connection setting"/>
+
+As can be observed, RudderStack dashboard provides the following fields for each of the above-mentioned settings:
+
+- **Default Value**: Enter the actual value for the field.
+- **RudderStack property name**: Enter a property name whose actual value will be taken from the payload and assigned to the field.
+
+Although these fields are optional, a higher priority is given to the **RudderStack property name** if both are specified. For example:
+- If you provide `new` in the **Default value** field for **X-S4-Status**, the output payload will contain `X-S4-Status: new`.
+- If you provide `alert_status` in the **RudderStack property name** field, then the **X-S4-Status** value is taken from `properties.alert_status`. Suppose the value for `properties.alert_status` is set to `resolved`, the output payload will contain `X-S4-Status: resolved`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:57]:
```suggestion
- If you set **RudderStack property name** for the **X-S4-Status** setting to `alert_status`, RudderStack looks for the event property `properties.alert_status`. Suppose the value for `properties.alert_status` is set to `resolved`, then the output payload sent to SIGNL4 will contain `X-S4-Status: resolved`.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+- **X-S4-ExternalID**: Use this field to pass the unique ID of a record if an event originates from that record in a third-party system. 
+<img src="../../assets/event-stream-destinations/signl4-conn-settings-5.png" alt="SIGNL4 connection setting"/>
+
+As can be observed, RudderStack dashboard provides the following fields for each of the above-mentioned settings:
+
+- **Default Value**: Enter the actual value for the field.
+- **RudderStack property name**: Enter a property name whose actual value will be taken from the payload and assigned to the field.
+
+Although these fields are optional, a higher priority is given to the **RudderStack property name** if both are specified. For example:
+- If you provide `new` in the **Default value** field for **X-S4-Status**, the output payload will contain `X-S4-Status: new`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:56]:
```suggestion
- If you provide `new` in the **Default value** field for the **X-S4-Status** setting, the event payload sent to SIGNL4 will contain `X-S4-Status: new`.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+- If you provide `alert_status` in the **RudderStack property name** field, then the **X-S4-Status** value is taken from `properties.alert_status`. Suppose the value for `properties.alert_status` is set to `resolved`, the output payload will contain `X-S4-Status: resolved`.
+
+If you provide values in both fields, the final output payload will contain `X-S4-Status: resolved`, as the **RudderStack property name** is given the higher priority.
+
+- **X-S4-Filtering**: Enable this setting and **Filter events/Signls** setting in the SIGNL4 dashboard to send an alert notification only if the input payload contains any keyword for a category. Refer to the <Link to="#signl-categories">Signl categories</Link> section to know more about keywords and category categories.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:61]: Can't find the **Filter events/Signls** setting in the below screenshot. Does toggling **Enable to add filter** enable this? 

If so, we can change the screenshot accordingly.

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+<img src="../../assets/event-stream-destinations/signl4-conn-settings-4.png" alt="SIGNL4 connection setting"/>
+
+<div class="infoBlock">
+Refer to the <a href="https://connect.signl4.com/webhook/docs/index.html">SIGNL4 documentation</a> to know more about these connection settings.
+</div>
+
+- **Event to title mapping**: Enter the event name and alert title you want to be mapped corresponding to each other. If you do not provide any value, the alert title is set as the event name.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:68]:
```suggestion
- **Event to title mapping**: This setting lets you map the RudderStack event name to a specific alert title. If you do not provide any value, RudderStack sets the alert title as the event name.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+
+## Signl categories
+
+In SIGNL4, you can use the categories to mark and tag Signls, for example, based on severity, type or relating IT systems, machines, sensors, and services. Refer to the [SIGNL4 documentation](https://account.signl4.com/manage/Category) to know more about different categories.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:72]:
```suggestion
In SIGNL4, you can use the categories to mark and tag Signls based on severity, type, or relating to IT systems, machines, sensors, and services. Refer to the [SIGNL4 documentation](https://account.signl4.com/manage/Category) to know more about the different categories.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+
+You can use the **X-S4-Service** setting in the RudderStack dashboard to specify the Signl category. If not provided, the keywords present in the payload decide the category of the alert. For example, an event payload containing keywords like `ambulance`, `doctor`, `hospital`, `nurse`, `surgery`, etc., is categorized as **Healthcare**.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:74]:
```suggestion
You can use the **X-S4-Service** setting in the RudderStack dashboard to specify the Signl category. If not provided, the keywords present in the payload decide the category of the alert. For example, an event payload containing keywords like `ambulance`, `doctor`, `hospital`, `nurse`, `surgery`, etc., is automatically categorized as **Healthcare**.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+
+<img src="../../assets/event-stream-destinations/categories-signl4.png" alt="SIGNL4 connection setting"/>
+
+## Track
+
+The <Link to="/event-spec/standard-events/track">`track`</Link> call can be used to push alerts for events with a particular alert title, alert message, or other additional details.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:80]:
```suggestion
You can use the <Link to="/event-spec/standard-events/track">`track`</Link> call to push alerts for events with a particular alert title, message, or other additional details.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+
+A sample `track` call is shown below:
+
+```javascript
+rudderanalytics.track(event, {
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:85]: Should `event` have a specific name? e.g. `"Security Alert"` cc @ujjwal-ab

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+  "status": "UP",
+  "message": "Agressive passenger",
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:87]:
```suggestion
  "message": "Aggressive passenger",
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+  "Source": "Gate agent",
+  "Type": "Security"
+});
+```
+
+You can pass any key-value pair in the `track` call; it is mapped as it is in the SIGNL4 destination.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:93]:
```suggestion
You can pass any key-value pair in the `track` call; RudderStack maps it as is in the SIGNL4 destination.
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+
+### Property mapping
+
+The following table lists the mappings between RudderStack and SIGNL4 properties:
+
+| RudderStack property | SIGNL4 property | Presence | Deescription |
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:99]:
```suggestion
| RudderStack property | SIGNL4 property | Presence | Description | 
```

File: docs/destinations/streaming-destinations/signl4.mdx [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+
+<img src="../../assets/event-stream-destinations/signl-dashboard-2.png" alt="SIGNL4 script url"/>
+
+### How can I create a member in the SIGNL4 team?
+
+To create a member in the SIGNL4 team, follow these steps:
+1. Log into your [SIGNL4 dashboard](https://connect.signl4.com/).
+2. Select **Users** > **Invite new user**, as shown:
+
+<img src="../../assets/event-stream-destinations/signl-dashboard-1.png" alt="SIGNL4 script url"/>
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:121]: IMO this is a very basic FAQ, do you think we can eliminate the image here and just list the basic steps? Saves us some space 😄

File: docs/destinations/streaming-destinations/signl4.mdx (hunk 18 above)
Comment (ujjwal-ab) [reply docs/destinations/streaming-destinations/signl4.mdx:85]: Ya, Let's add a specific event name. Let's add `Security Alert`.

Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/signl4.mdx:85]:
```suggestion
rudderanalytics.track("Security Alert", {
```

File: docs/destinations/streaming-destinations/signl4.mdx (hunk 13 above)
Comment (ishachopra0401) [reply docs/destinations/streaming-destinations/signl4.mdx]: nope, that setting is in the SIgnl4 dashboard, as mentioned. Also, there is a screenshot in **Signl categories** section and a reference is added here.

**Review by ameypv-rudder — APPROVED** [review]

**Pull request merged**